### PR TITLE
feat: Add Get role

### PR DESCRIPTION
**Pull request by jacks0n9**
# Description

## Breaking Change?

no

## Benchmarks

# Checklist:

- [x] I have performed a self-review of my own code
- [x] Commented complex situations or referenced the discord documentation
- [x] Updated documentation
- [x] Added/Updated unit tests
- [ ] Added/Updated benchmarks (if this is a performance critical component)

### CODE REVIEW

File: guild.go [hunk 1]
```diff
@@ -747,7 +747,7 @@ type GuildQueryBuilder interface {
 	// GetRoles
 	// TODO: For GetRoles, it might sense to have the option for a function to filter before each role ends up deep copied.
 	// TODO-2: This could be much more performant in larger guilds where this is needed.
-	// TODO-3: Add GetRole.
```
Comment (andersfylling) [inline guild.go:750]: I see my comment has misguided you, that's on me. Could you adjust this so it's `Client.Guild(id).Role(id).Get()` instead?

Comment (andersfylling) [reply guild.go:750]: Essentially you just move the new method to the `GuildRoleQueryBuilder` interface instead, and name it `Get`.

File: guild.go [hunk 2]
```diff
@@ -1191,6 +1190,21 @@ func (g guildQueryBuilder) GetRoles() ([]*Role, error) {
 	return getRoles(r.Execute)
 }
 
+// Search through GetRoles to find a role
+func (g guildRoleQueryBuilder) Get(ID Snowflake) (*Role, error) {
```
Comment (andersfylling) [inline guild.go:1194]: remove the ID parameter, and use `g.roleID` instead

File: role.go [hunk 3]
```diff
@@ -78,10 +78,9 @@ func (r *Role) SetGuildID(id Snowflake) {
 type GuildRoleQueryBuilder interface {
 	WithContext(ctx context.Context) GuildRoleQueryBuilder
 	WithFlags(flags ...Flag) GuildRoleQueryBuilder
-
 	Update(params *UpdateRole) (*Role, error)
 	Delete() error
-
+	Get(ID Snowflake) (*Role, error)
```
Comment (andersfylling) [inline role.go:83]: remove parameter

File: guild.go [hunk 4]
```diff
@@ -1191,6 +1190,21 @@ func (g guildQueryBuilder) GetRoles() ([]*Role, error) {
 	return getRoles(r.Execute)
 }
 
+// Search through GetRoles to find a role
+func (g guildRoleQueryBuilder) Get() (*Role, error) {
+	guild := g.client.Guild(g.gid)
+	roles, err := guild.GetRoles()
+	if err != nil {
+		return &Role{}, err
```
Comment (andersfylling) [inline guild.go:1198]: return `nil, err` instead, no need to instantiate a empty role here. Same for the bottom statement

File: guild.go [hunk 5]
```diff
@@ -1191,6 +1190,21 @@ func (g guildQueryBuilder) GetRoles() ([]*Role, error) {
...
+	}
+	for _, role := range roles {
+		if role.ID == g.roleID {
+			return role, nil
+		}
+	}
+	return &Role{}, nil
```
Comment (andersfylling) [inline guild.go:1205]: This should return an error saying "role not found" or something.

Comment (jacks0n9) [reply guild.go]: done!

File: guild.go (hunk 4 above)
Comment (jacks0n9) [reply guild.go]: done

**Pull request merged**